openpype/pipeline/workfile/lock_workfile.py

```diff
@@ -0,0 +1,67 @@
+import os
+import json
+from uuid import uuid4
+from openpype.lib.pype_info import get_workstation_info
+
+
+def _read_lock_file(lock_filepath):
+    with open(lock_filepath, "r") as stream:
+        data = json.load(stream)
+    return data
+
+
+def _get_lock_file(filepath):
+    return filepath + ".lock"
+
+
+def is_workfile_locked(filepath):
+    lock_filepath = _get_lock_file(filepath)
+    if not os.path.exists(lock_filepath):
+        return False
+    return True
+
+
+def is_workfile_locked_for_current_process(filepath):
+    if not is_workfile_locked():
+        return False
+
+    lock_filepath = _get_lock_file(filepath)
+    process_id = os.environ["OPENPYPE_PROCESS_ID"]
+    data = _read_lock_file(lock_filepath)
+    return data["process_id"] == process_id
+
+
+def delete_workfile_lock(filepath):
+    lock_filepath = _get_lock_file(filepath)
+    if not os.path.exists(lock_filepath):
+        return
```

Comment:
No need to check this since `is_workfile_locked_for_current_process` already checks that too. Right?

Reply:
I didn't use this function in other script. I guess it wont be problem if it gets removed.